sys-cluster/opencoarrays/opencoarrays-2.10.0.ebuild

```diff
@@ -0,0 +1,35 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+CMAKE_MAKEFILE_GENERATOR=emake
+FORTRAN_STANDARD="2003"
+
+inherit cmake fortran-2
+
+MY_PN="OpenCoarrays"
+
+DESCRIPTION="A parallel application binary interface for Fortran 2018 compilers"
+HOMEPAGE="http://www.opencoarrays.org/"
+SRC_URI="https://github.com/sourceryinstitute/${MY_PN}/releases/download/${PV}/${MY_PN}-${PV}.tar.gz"
+
+S="${WORKDIR}/${MY_PN}-${PV}"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+# Tests fail with FEATURES="network-sandbox" for most versions of openmpi and mpich it with error:
+# "No network interfaces were found for out-of-band communications.
+#  We require at least one available network for out-of-band messaging."
+IUSE="test"
+PROPERTIES="test_network"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	|| ( >=sys-cluster/openmpi-4.1.2[fortran] >=sys-cluster/mpich-3.4.3[fortran,mpi-threads,threads] )
```

Comment:
can't you use virtual/mpi?

Reply:
No. I discussed it with one of gentoo developers and he offered to specify explicit implementation as MPI 3.1 support is required by package while not all virtual/mpi packages provides it.

Reply:
maybe virtual/mpi can be versioned?

Reply:
Agree. But I need to recheck again what MPI 3.1 feature was required and test against nullmpi and nativempi.

Reply:
@Alessandro-Barbieri maybe `virtual/mpi[-nullmpi]` is suitable?

But I'm confused that `nullmpi? ( sys-cluster/native-mpi )` within virtual/mpi but not `nullmpi? ( sys-cluster/nullmpi )`.
@thesamesam is it alright?

But the minimal supported version of MPI by sys-cluster/native is still under question for me :(

Reply:
@Alessandro-Barbieri there is another one problem: package could require `sys-cluster/mpich[fortran,mpi-threads,threads]` while `virtual/mpi` doesn't provide `mpi-threads` USE-flag but without it the tests are broken.

I don't know the details why `treads` and `mpi-threads` are split in `sys-cluster/mpich`.